test/jtreg-ext/requires/VMProps.java

```diff
@@ -715,6 +722,46 @@ private String jdkContainerized() {
         return "" + "true".equalsIgnoreCase(isEnabled);
     }
 
+    private String packagedModules() {
+        // Some jlink tests require packaged modules being present (jmods).
+        // For a runtime linkable image build packaged modules aren't present
+        try {
+            Path jmodsDir = Path.of(System.getProperty("java.home"), "jmods");
+            if (jmodsDir.toFile().exists()) {
+                return Boolean.TRUE.toString();
+            } else {
+                return Boolean.FALSE.toString();
+            }
+        } catch (Throwable t) {
+            return Boolean.FALSE.toString();
+        }
+    }
+
+    private String runtimeLinkable() {
+        // jdk.jlink module has the following resource indicating a runtime-linkable
+        // image. It's the diff file for runtime linking of the java.base module.
+        String linkableRuntimeResource = "jdk/tools/jlink/internal/runtimelink/diff_java.base";
```

Comment:
Can this access `LinkableRuntimeImage::DIFF_PATTERN` via qualified exports?

Reply:
I don't see how, but I may be missing something. `VMProps` isn't in any named module is it? Is it worth sprinkling `jdk.jlink`'s `module-info.java` classes with those exports for test-only purposes, though? Is there precedent that I can look at?

It appears this hard-coded resource seems the lesser of the two evils to me. Thoughts?

Reply:
With the latest changes there is no `VMProps` extension to determine a run-time image link capable JDK anymore. So this is gone.